docs/contact_plugin.rst

```diff
@@ -1,28 +1,92 @@
+
+Contact energy is constantly re-calculated each time a cell's surface changes.
+
+We define contact energy as a matrix that compares each cell type against each other cell type. 
+In CC3DML, try changing the contact energy between CellA and CellB to ``2``. 
+Keep everything else the same, then click Play |Play|  to see the result. 
+
+.. code-block:: xml
+
+    <Plugin Name="Contact">
+        <Plugin Name="Contact">
+        <Energy Type1="CellA" Type2="CellA">10</Energy>
+        <Energy Type1="CellB" Type2="CellB">10</Energy>
+        <Energy Type1="CellA" Type2="CellB">2</Energy> <!--Here!-->
+        <Energy Type1="CellA" Type2="Medium">10</Energy>
+        <Energy Type1="CellB" Type2="Medium">10</Energy>
+        <Energy Type1="Medium" Type2="Medium">10</Energy> <!--Medium-Medium energy doesn't really matter-->
+        <NeighborOrder>2</NeighborOrder>
+    </Plugin>
+
+This should stimulate cell mixing, the opposite of `cell Sorting <example_cell_sorting.html>`_. 
+
+
+What Does Energy Mean?
+-------------------------------------------------
+
+When tweaking your parameters, it's most important to pay attention to their **relative values**. 
+
+However, the actual value of each parameter—whether you set ``10`` or ``1``—affects the contact behaviors, too, because of *temperature*. 
+
+Each time a cell flexes its surface, a decision is made about which pixels in the lattice will be copied over to new lattice sites. 
+We discussed this in the `Potts page <potts.html>`_. 
+
+The probability of accepting a lattice site copy is:
 
 .. math::
     :nowrap:
 
     \begin{eqnarray}
-        E_{adhesion} = \sum_{i,j,neighbors} J\left ( \tau_{\sigma(i)},\tau_{\sigma(j)} \right )\left ( 1-\delta_{\sigma(i), \sigma(j)} \right )
+        P = e^{-\bigtriangleup H/T}
     \end{eqnarray}
 
-where ``i`` and ``j`` label two neighboring lattice sites :math:`\sigma`'s denote cell
-Ids, :math:`\tau`'s denote cell types .
+where :math:`\bigtriangleup H = H_f - H_i`. 
+:math:`H_f` is the new energy after a potential lattice site copy. 
+:math:`H_i` is the current energy of the system at step *i*. 
+The new energy, :math:`H_i`, may be higher, lower, or the same after a site copy. 
+Meanwhile, :math:`T` is the temperature (or fluctuation amplitude *or* membrane fluctuation level *or* noise level).
+A high temperature lowers the probability that site copy attempts will be made. 
+
+
+Calculating Contact Energy
+-------------------------------------------------
+
+The explicit formula for contact energy is:
+
+.. math::
+    :nowrap:
+
+    \begin{eqnarray}
+        E_{contact} = \sum_{i,j,neighbors} J\left ( \tau_{\sigma(i)},\tau_{\sigma(j)} \right )\left ( 1-\delta_{\sigma(i), \sigma(j)} \right )
+    \end{eqnarray}
 
+where ``i`` and ``j`` label two neighboring lattice sites :math:`\sigma`'s denote cell
+IDs, and each :math:`\tau` denotes a cell type. 
+Note that ``i`` and ``j`` are are at separate coordinates; for example, ``i = (0,4)`` and ``j = (1,4)``. 
```

Comment:
I had a hard time understanding this formula, so I added the a few other formulae in this doc. What is delta in this formula? Why do we subtract from 1?
https://pythonscriptingmanual.readthedocs.io/en/latest/contact_plugin.html